app/api/domains/cho.py

```diff
@@ -712,14 +719,9 @@ async def login(
     db_country = user_info.pop("country")
 
     if not ip.is_private:
-        if app.state.services.geoloc_db is not None:
-            # good, dev has downloaded a geoloc db from maxmind,
-            # so we can do a local db lookup. (typically ~1-5ms)
-            # https://www.maxmind.com/en/home
-            geoloc = app.state.services.fetch_geoloc_db(ip)
-        else:
-            # bad, we must do an external db lookup using
-            # a public api. (depends, `ping ip-api.com`)
+        geoloc = app.state.services.fetch_geoloc_nginx(ip, headers)
```

Comment:
why does login not check for cloudflare headers? i remember people mentioning they had to do some weird configs on cloudflare for it to work. i think this might be the reason why. also if we decouple the IP lookup strategy logic from ``domains/osu.py`` it could be reused here.